src/internal/storage/Compression.h

```diff
@@ -0,0 +1,593 @@
+    
+}
+
+/// Compression algorithm is based on
+/// <a href="https://en.wikipedia.org/wiki/Huffman_coding#Basic_technique">huffman coding</a>
+/// and is separated in 2 parts:
+/// <br> <br>
+/// <h2>Part 1</h2>
+/// <ol type = "1">
+///     <li>Size information</li>
+///     <li>Counting usage frequency of unique bytes and unique byte count</li>
+///     <li>Creating the base of the translation array</li>
+///     <li>Creating the translation tree inside the translation array by weight distribution</li>
+///     <li>Adding strings from top to bottom to create translated versions of unique bytes</li>
+/// </ol>
+/// <br>
+/// <h2>Part 2</h2>
+/// <ul>
+///     <li>first (one byte) -> symbols</li>
+///     <li>second (bit groups)
+///         <ul>
+///             <li>(8 bits) -> current unique byte</li>
+///             <li>(8 bits) -> length of the transformation</li>
+///             <li>(bits) -> transformation code of that unique byte</li>
+///         </ul>
+///     </li>
+///     <li>third (2 bytes)**1** -> file_count (inside the current folder)
+///     <li>fourth (1 bit)**2** -> file or folder information -> folder(0), file(1)</li>
+///     <li>fifth (8 bytes) -> size of current input_file (IF FILE)
+///     <li> sixth (bit group)
+///         <ul>
+///             <li>(8 bits) -> length of current input_file's or folder's name</li>
+///             <li>(bits) -> transformed version of current input_file's or folder's name</li>
+///         </ul>
+///     </li>
+///     <li>seventh (a lot of bits) -> transformed version of current input_file (IF FILE)</li>
+/// </ul>
+/// **1** groups from fifth to eighth will be written as much as file count in that folder <br>
+///    (this is argument_count-1(argc-1) for the main folder) <br>
+/// **2** whenever we see a new folder we will writeFromCh seventh then start writing from fourth to eighth
+class Compress {
+public:
+        std::vector<std::string> files;
+        
+/// \brief Constructor of the compression class with which you can compress provided files
+///
+/// \param argc - number of files for compress
+/// \param argv - path's to files for compress
+        Compress(const int argc, const char *argv[])
+        {
+                files.reserve(argc - 1);
+                for (int i = 1; i < argc; ++i)
+                        files.emplace_back(argv[i]);
+                
+                if (argc == 2) {
+                        compressed_name = files[0];
+                        compressed_name += ".huff";
+                } else
+                        compressed_name = "bundle.huff";
+        }
+
+/// \brief checks if test condition is false or true
+///
+/// \param test - false or true. If true -> print the error, else continue
+/// \param message - message that represents the error
+/// \param fd - file descriptor for closing
+/// \param ... - arguments for error printing
+        static void check(int fd, bool test, const char *message, ...)
```

Comment:
I remove it. It's misunderstanding on my part because I use it only once at this point. In previous commits I use it for error handling in recursive function(countFolderBytesFreq and countFileBytesFreq) to iterate through directories, but now I use std::filesystem ;)